modules/loki/main.tf

```diff
@@ -24,3 +43,66 @@ resource "helm_release" "loki" {
     )
   ]
 }
+
+
+module "loki_bucket" {
+  source  = "dasmeta/s3/aws"
+  version = "1.3.1"
+
+  name = local.s3_bucket_name
+}
+
+module "loki_iam_eks_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "5.53.0"
+
+  count = local.create_loki_role ? 1 : 0
+
+  role_name = "loki-role"
```

Comment:
have this name dynamic or configurable to not have collisions when we will have need to enable loki in same aws account different clusters,
maybe you can use cluster name in name and random string to have it dynamic/unique